packages/react/src/__tests__/forwardRef-test.js

```diff
@@ -242,6 +242,33 @@ describe('forwardRef', () => {
     ).toErrorDev(
       'Warning: Failed prop type: The prop `required` is marked as required in ' +
         '`Foo`, but its value is `undefined`.\n' +
+        '    in Unknown (at **)',
+    );
+  });
+
+  it('should honor a displayName in stacks if set on the inner function', () => {
```

Comment:
This will make some people unhappy. Can we make it work with the outer one too in DEV? For all the wrappers: lazy, memo, forwardRef.

Reply:
What is worse is that Firefox doesn't respect displayName in stacks at all so none of them will be respected in Firefox. However, that's also true for regular error stacks so they should just fix that there. The goal is to be no better or worse than native stacks - and ideally just like native stacks.

I don't really want to add code to parse and try to piece together a line by trying to inject a name in the middle of the browser native format somehow.

What we could do is transfer the displayName onto the inner function if one is not already defined. This would be invasive in the sense that it's mutating an object not owned by React.

Reply:
I added a forwarding setter in DEV. https://github.com/facebook/react/pull/18495/commits/9054ad29cd844b0dba230dce0572a52dd75c8a53

Not for lazy though because you're suppose to use that on the consuming side, and that's not where you specify things like propTypes or displayName. Not sure they should really be used with forwardRef/memo neither. It should really be the inner function that you name.

Reply:
Ofc we had a test that reused a component. This trick doesn't work in this case. I think it's unusual though. https://github.com/facebook/react/pull/18495/commits/544983849e32e4a2c58ac775880346378323fb18